lib/Local.js

```
@@ -26,6 +26,52 @@ function Local(){
   this.errorRegex = /\*\*\* Error: [^\r\n]*/i;
   this.doneRegex = /Press Ctrl-C to exit/i;
 
+  this.startSync = function(options) {
+    this.userArgs = [];
+    var that = this;
+    this.addArgs(options);
+
+    if(typeof options['onlyCommand'] !== 'undefined')
+      return;
+
+    const binaryPath = this.getBinaryPath();
+    that.binaryPath = binaryPath;
+    childProcess.exec('echo "" > ' + that.logfile);
+    that.opcode = 'start';
+    if(!this.binaryPath){
+      return new LocalError('Couldn\'t find binary file');
+    }
+    try{
+      const obj = childProcess.spawnSync(that.binaryPath, that.getBinaryArgs());
+      this.tunnel = {pid: obj.pid};
+      var data = {};
+      if(obj.stdout.length > 0)
+        data = JSON.parse(obj.stdout);
+      else if(obj.stderr.length > 0)
+        data = JSON.parse(obj.stderr);
+      else
+        return new LocalError('No output received');
+      if(data['state'] != 'connected'){
+        return new LocalError(data['message']['message']);
+      } else {
+        that.pid = data['pid'];
+        that.isProcessRunning = true;
+        return;
+      }
+    }catch(error){
+      console.error('Error while trying to execute binary', error);
+      if(that.retriesLeft > 0) {
+        console.log('Retrying Binary Download. Retries Left', that.retriesLeft);
+        that.retriesLeft -= 1;
+        fs.unlinkSync(that.binaryPath);
+        delete(that.binaryPath);
+        return that.startSync(options);
+      } else {
+        return new LocalError(error.toString());
```

@07souravkunda This should throw an error instead of returning it ~
@francisf